### ISSUE TITLE
can not use Command+S to save files after upgrading to 0.9.9.0

### ISSUE THREAD

**Issue by leewi9** [typora/typora-issues#106]
pls see this pic and gif for details.

![snip20160404_6](https://cloud.githubusercontent.com/assets/10832544/14237881/817258f2-fa5c-11e5-88e2-40458750f288.png)

![untitled](https://cloud.githubusercontent.com/assets/10832544/14237869/4feab55e-fa5c-11e5-82fc-3ed124745e49.gif)

**Comment by abnerlee** [comment]
This works fine on my Mac. Does this bug always happens or happens occasionally?

Could restart Typora fix this

**Comment by leewi9** [comment]
After restart, it works fine now. It happened two times, yesterday and today.

**Comment by waahto** [comment]
Couldn't replicate either.

**Comment by leewi9** [comment]
What do you mean ?

-- 
leewi9
Sent with Airmail

On April 4, 2016 at 4:51:26 PM, waahto (notifications@github.com) wrote:

Couldn't replicate either.

**Comment by waahto** [comment]
Sorry for being too brief :) After some testing, I couldn't get Typora to *not* save the edited document.

**Comment by leewi9** [comment]
Ok, maybe it’s just an occasional issue.

-- 
leewi9
Sent with Airmail

On April 4, 2016 at 5:16:25 PM, waahto (notifications@github.com) wrote:

Sorry for being too brief :) After some testing, I couldn't get Typora to not save the edited document.

**Comment by abnerlee** [comment]
@leewi9 did you restart typora after upgrading to latest version when this bug occurs?

**Comment by leewi9** [comment]
Sorry I didn’t notice.

I was checking the update and install updates, then click ... maybe relaunch?


-- 
leewi9
Sent with Airmail

On April 5, 2016 at 5:41:38 PM, AbnerLee (notifications@github.com) wrote:

@leewi9 did you restart typora after upgrading to latest version when this bug occurs?

**Comment by waahto** [comment]
I got it! Don't know how, but it did happen. I have a file currently open and I'm experiencing exactly the same behaviour as leewi9. I have no idea how this happened or how to replicate this.

- I opened the file in TextMate simultaneously. Text mate is set up so that it automatically reopens the file if the file changes. When I type something in Typora, it also updates the changes to TextMate also. This means that Typore does save the file — even without hitting `cmd+s` or the menu(!)
- Finder also shows that the time stamp of the file is changing whenever something is typed into it.
- When I open an existing file, it behaves similarly.
- When I open a *new* file, it behaves similarly, when closing it, it asks where to save it — as is normal.
- Flipping to source code mode and back or changing the theme didn't help (had to try...)

I'm not planning to restart Typora for a while so I can try stuff out if somebody has suggestions.

**Comment by abnerlee** [comment]
@waahto Thanks for the info.

But honestly I still don't know the reason for this And didn't got the chance to reproduce it.

If you still got typora open, could you try if typora is working properly? 
Like input `# header`, and see if the outline is updated or not.
Or try if switch between typora windows would solve this.
Or anything from menu or anything else just to see if there's some other strong behaviors which may give us some clue~

PS, did you restart Typora manually after you upgrade to latest version?

**Comment by DanBennettUK** [comment]
Also confirm the permissions of the file you're editing - that you're use
is actually allowed to write to it. It looks more like it's not allowed to
write. (Also try opening Typora as Sudo for testing)

**Comment by waahto** [comment]
I tested different markdown elements, outline, tried all kinds of gimmicks at the beginning of the file, loaded and created files, closed all open files (leaving Typora open) but everything else works as usual. (And yes, permissions are ok).

I'm definitely not a pro on debugging but I noticed that the OS doesn't know that x.md that I got open in Typora is actually open in Typora. On activity monitor there's the info button and after I click it, there's a tab called "open files and ports" but there's no x.md there. When I open y.md in Typora, it does pop in to that list. When I type a character to y.md to change it's status to "edited" in Typora, it disappears from the list. Did I make sense?

**Comment by leewi9** [comment]
Hmm, that’s quite wired, maybe should submit a ticket to Apple?

-- 
leewi9
Sent with Airmail

On April 7, 2016 at 1:26:18 PM, waahto (notifications@github.com) wrote:

I tested different markdown elements, outline, tried all kinds of gimmicks at the beginning of the file, loaded and created files, closed all open files (leaving Typora open) but everything else works as usual. (And yes, permissions are ok).

I'm definitely not a pro on debugging but I noticed that the OS doesn't know that x.md that I got open in Typora is actually open in Typora. On activity monitor there's the info button and after I click it, there's a tab called "open files and ports" but there's no x.md there. When I open y.md in Typora, it does pop in to that list. When I type a character to y.md to change it's status to "edited" in Typora, it disappears from the list. Did I make sense?

**Comment by waahto** [comment]
Hmm... It also appears that closing a file in Typora "hides" it rather than closes it. It stays on the "open files and ports" list and I get scolded by other programs if trying to access that file.

**Comment by abnerlee** [comment]
@waahto Thanks for the report. This should be a new bug since 0.9.9.0. We will fix this in next version up.

**Comment by leewi9** [comment]
@abnerlee Hi, this bug happens again after I upgraded to the newest version 0.9.9.2.1.

**Comment by abnerlee** [comment]
@leewi9 maybe it is related the auto-relaunch after upgrade.

Have you do restarting manually after upgrade?

**Comment by leewi9** [comment]
Yes, maybe.

I didn't restart manually after upgrade, let me try again.


2016年4月19日 +0800 23:13 AbnerLee<notifications@github.com>，写道：

**Comment by abnerlee** [comment]
It's related to menu validation, should have been fixed in 0.9.9.2.4. 

But I'm not 100% sure since I didn't reproduce the bug.

Please re-open this issue if it happens again.
